### PR TITLE
wal-g postgres wal-push command

### PR DESCRIPTION
**Pull request by sergey-arefev**
the first client implementation to send commands to wal-g in daemon mode

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# WAL-G daemon client
+
+lightweight client for [WAL-G](https://github.com/wal-g/wal-g/) daemon
```
Comment (usernamedt) [inline README.md:3]: We can provide a direct link to the functionality we're referring to (https://github.com/wal-g/wal-g/blob/master/docs/PostgreSQL.md#daemon)

File: cmd/walg-daemon-client/main.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+const (
+	cmdUsageMessage = `walg-daemon-client is lightweight client for WAL-G daemon
+
+Usage:
+	walg-daemon-client socket command [filename]
+
+Arguments:
+	socket		- name of unix socket to communicate with wal-g daemon
+	command		- command to send to the daemon: wal-push, wal-fetch
```
Comment (usernamedt) [inline cmd/walg-daemon-client/main.go:20]: I would say `socket command [command_args]` would be more universal

File: pkg/common/common.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+func getMessage(messageType SocketMessageType, messageBody string) []byte {
+	res := binary.BigEndian.AppendUint16(messageType.ToBytes(), uint16(len(messageBody)+3))
+	if len(messageBody) > 0 {
+		return append(res, []byte(messageBody)...)
+	}
+	return res
+}
+
+func SendCommand(opts *RunOptions) {
```
Comment (usernamedt) [inline pkg/common/common.go:46]: I guess this logic should belong to the `internal` package

File: pkg/common/common.go [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+	dialer := net.Dialer{}
+	daemonAddr := net.UnixAddr{Name: opts.SocketName, Net: "unix"}
+	socketConnection, err := dialer.DialContext(ctx, "unix", daemonAddr.String())
+	if err != nil {
+		log.Fatalf("unix socket dial error: %v\n", err)
+	}
+	defer socketConnection.Close()
+	err = socketConnection.SetDeadline(time.Now().Add(DaemonSocketOperationTimeout))
+	if err != nil {
+		log.Fatalf("unix socket set deadline error: %v\n", err)
```
Comment (usernamedt) [inline pkg/common/common.go:59]: we need to handle errors on the top level (e.g. `return err` instead of fatal)

File: pkg/common/common.go [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+	"log"
+	"net"
+	"time"
+)
+
+const (
+	DaemonSocketConnectionTimeout = time.Second
+	DaemonSocketOperationTimeout  = 5 * time.Second
+
+	CheckType    SocketMessageType = 'C'
```
Comment (usernamedt) [inline pkg/common/common.go:15]: I propose to extract these common message types [here](https://github.com/wal-g/wal-g/tree/master/pkg) and import it instead.

**Review by usernamedt — CHANGES_REQUESTED** [review]

File: cmd/walg-daemon-client/main.go (hunk 2 above)
Comment (sergey-arefev) [reply cmd/walg-daemon-client/main.go]: changed to flags

File: cmd/walg-daemon-client/main.go [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+	command			- command to send to the daemon: wal-push, wal-fetch
+
+Flags:
+`
+)
+
+func parseArgs(args []string) (*common.RunOptions, *flag.FlagSet, error) {
+	opts := &common.RunOptions{}
+	fs := flag.NewFlagSet("", flag.ContinueOnError)
+	fs.StringVar(&opts.FileName, "file", "", "WAL file name (required for postgres specific commands: wal-push, wal-fetch)")
```
Comment (usernamedt) [inline cmd/walg-daemon-client/main.go:32]: wal-push command requires single arg but wal-fetch require two args, so this single flag won't work. I suggest to stick with approach with args count varying between different commands, e.g.:
```
walg-daemon-client socket wal-push wal_path [flags]
walg-daemon-client socket wal-push wal_name destination_path [flags]
```

**Review by usernamedt — APPROVED** [review]

**Pull request merged**